### PR TITLE
Implement security / content restrictions for preview attachment WebView

### LINKED ISSUE #1213: implement security / content restrictions for preview attachment WebView
It seems there is no 1 place where you can find all possible security settings for `WKWebView`, but it's quite customizable:
- for disabling navigation to external links we can implement `func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void)` method of `WKNavigationDelegate` in `AttachmentViewController`
- for disabling JS we should use `allowsContentJavaScript` property of `WKWebpagePreferences`
- loading of external resources can be disabled by using `WKContentRuleList` (https://stackoverflow.com/a/60864550/3091318)

_Originally posted by @sosnovsky in https://github.com/FlowCrypt/flowcrypt-ios/issues/1205#issuecomment-991928087_

### PR DESCRIPTION
**Pull request by ivan-ushakov**
close #1213

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests will be added later (issue #...)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by ivan-ushakov** [comment]
Title: Cross-site scripting in attachment preview

Simulator Screen Shot - iPhone 11 Pro - 2021-12-19 at 14 46 11

No `hacked by mart` text

**Comment by ivan-ushakov** [comment]
Title: Arbitrary HTTP GET request in SVG attachment preview

Simulator Screen Shot - iPhone 11 Pro - 2021-12-19 at 14 52 33

No flower

Original SVG:
```
<?xml version="1.0" encoding="UTF-8" standalone="no"?><svg xmlns:svg="http://www.w3.org/2000/svg" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="200" height="200"><image height="200" width="200" xlink:href="https://upload.wikimedia.org/wikipedia/commons/3/3f/JPEG_example_flower.jpg" /></svg>
```

In macOS preview you can see flower.

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift [hunk 1]
```diff
@@ -99,22 +131,40 @@ extension AttachmentViewController: WKNavigationDelegate {
...
 }
 
 private extension AttachmentViewController {
 
+    private func setContentRules() async throws {
+        guard let list = try await WKContentRuleListStore.default().compileContentRuleList(
+            forIdentifier: "blockRules",
+            encodedContentRuleList: encodedContentRules
+        ) else {
+            throw AppErr.general("Could not produce a data URL to preview this file")
```
Comment (ivan-ushakov) [inline FlowCrypt/Controllers/Attachment/AttachmentViewController.swift:153]: Should be another error text

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift [hunk 2]
```diff
@@ -25,6 +24,20 @@ final class AttachmentViewController: UIViewController {
...
+        let preferences = WKWebpagePreferences()
+        preferences.allowsContentJavaScript = false
+
+        let configuration = WKWebViewConfiguration()
+        configuration.defaultWebpagePreferences = preferences
+
+        let view = WKWebView(frame: .zero, configuration: configuration)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.navigationDelegate = self
+        view.backgroundColor = .backgroundColor
```
Comment (sosnovsky) [inline FlowCrypt/Controllers/Attachment/AttachmentViewController.swift:37]: Let's add `view.allowsLinkPreview = false` here, as currently long pressing on link will open preview, which can contain malicious content.

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift [hunk 3]
```diff
@@ -99,22 +131,40 @@ extension AttachmentViewController: WKNavigationDelegate {
...
 
+    private func setContentRules() async throws {
+        guard let list = try await WKContentRuleListStore.default().compileContentRuleList(
+            forIdentifier: "blockRules",
+            encodedContentRuleList: encodedContentRules
+        ) else {
+            throw AppErr.general("Could not produce a data URL to preview this file")
+        }
+
+        webView.configuration.userContentController.removeAllContentRuleLists()
```
Comment (sosnovsky) [inline FlowCrypt/Controllers/Attachment/AttachmentViewController.swift:156]: Is it necessary to remove all previous content rules lists?
What if user has some content-blocker installed with custom block lists, will they be removed too?

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift (hunk 1 above)
Comment (sosnovsky) [reply FlowCrypt/Controllers/Attachment/AttachmentViewController.swift:153]: We can use `do/catch` here and use original error:
```swift
do {
    let list = try await WKContentRuleListStore.default().compileContentRuleList...
    ....
} catch {
    throw error
}
```

**Review by sosnovsky — CHANGES_REQUESTED** [review]

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift (hunk 3 above)
Comment (tomholub) [reply FlowCrypt/Controllers/Attachment/AttachmentViewController.swift:156]: Not sure, but I'd rather we have more control, until the details are more understood. Which would mean remove all first.

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift (hunk 2 above)
Comment (tomholub) [reply FlowCrypt/Controllers/Attachment/AttachmentViewController.swift:37]: Excellent observation

File: FlowCrypt/Controllers/Attachment/AttachmentViewController.swift (hunk 1 above)
Comment (ivan-ushakov) [reply FlowCrypt/Controllers/Attachment/AttachmentViewController.swift]: This will not work here because we get optional and have to log error anyway.

**Review by sosnovsky — APPROVED** [review]

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**